libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java

```diff
@@ -59,20 +63,36 @@ public double zoom(RouteInformation routeInformation) {
     return CAMERA_ZOOM;
   }
 
+  @Override
+  public List<Point> overview(RouteInformation routeInformation) {
```

Comment:
IMO `overview` is getting too long.
What about extracting blocks of code into well-named private methods and simplifying conditional expressions somehow?

We should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).